src/app/(auth)/subscriber/edit/page.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import SubscriberForm from '@/components/Subscriber/SubscriberForm';
+import { UserGroupIcon } from '@heroicons/react/24/outline';
+import { updateSubscriber, getSubscriber } from '@/apis/subscriber';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+import { useRouter, useSearchParams } from 'next/navigation';
+import { extractValuesFromOptions } from '@/utils/common';
+
+const Page = () => {
+  const searchParams = useSearchParams();
+  const id = searchParams.get('id');
+  const router = useRouter();
+  const [subscriber, setSubscriber] = useState({});
+
+  useEffect(() => {
+    const fetchSubscriber = async () => {
+      try {
+        const { data } = await getSubscriber(id);
+        setSubscriber(data);
+      } catch (error: any) {
+        toast.error(error.response.data.message);
+      }
+    };
+    fetchSubscriber();
+  }, []);
+
+  const onSubmit = async (values: any) => {
+    values.tags = extractValuesFromOptions(values.tags);
```

Comment:
what's the reason for not putting this line into SubscriberForm and have it repeated twice in create and edit pages?

My thought has been that we do all data transformation & functionality specific to the form and submitted values inside the SubscriberForm and keep the pages only to make API requests and redirection.

So I think the SubscriberForm should send formatted data as the output for submit.

Reply:
Got it and make sense your point here. Made the changes accordingly. Can you please take a look once.